### DISCUSSION TITLE
Control who can create an account

### DISCUSSION THREAD

**Discussion by JonasBarcat** [pglombardo/PasswordPusher#356] [Q&A] [answered]
Hello! Is there any way to control who can create and account, I mean, to control who can sign up in the application?

**Comment by pglombardo** [comment]
Hi @JonasBarcat,

There is no way to control this right now.  But if we could, how would you do this?

1. Limit to an email domain?  e.g. only users with `@pwpush.com` emails can sign up
2. IP block? (I would use a firewall in this case)
3. Disable registrations and allow only an admin type user to do specific invites?

`1` might be the easiest but which is best depends on the use case/need.

I've had `3` on the list for a while but it's a big job.  The plan was to do this at the same time of adding teams and a group policy.

**Reply by JonasBarcat** [reply]
Thank you for you support!

**Reply by pglombardo** [reply]
Hi @JonasBarcat - just a late update that we now support limiting signups by email domain.  e.g. only allow users with your company domain to sign up.  Hope this helps!

__August 2023 Update__: We now support limiting sign ups to specific email domains.  See `PWP__SIGNUP_EMAIL_REGEXP` in the [Configuration document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins).

**Comment by pglombardo** [comment, marked as answer]
Hi @JonasBarcat - updating the answer with the latest information:

To control account creation, [enable logins](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins) and set `PWP__ALLOW_ANONYMOUS` to false.  This blocks all anonymous users from creating text, file or URL pushes.

Then once all of your teammate accounts are made, set `PWP__DISABLE_SIGNUPS` which will disable new account registration entirely.

Anonymous users will still be able to receive/retrieve secret URLs that are given to them.

More details in the [Configuration.md document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins).

__August 2023 Update__: We now support limiting sign ups to specific email domains.  See `PWP__SIGNUP_EMAIL_REGEXP` in the [Configuration document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins).

I hope I explained this well and it helps out.
